gst/nnstreamer/tensor_query/tensor_query_common.h

```diff
@@ -76,31 +78,63 @@ typedef struct
 
 /**
  * @brief generate unique id.
- * @return unique id if OK, 0 if error
+ * @return int64_t* unique id if OK, NULL if error
  */
-extern uint64_t
-nnstreamer_query_request_id (const char *ip, uint32_t port, int is_client);
+extern int64_t *
+nnstreamer_query_request_id (const char * server_host, uint32_t server_port, const char * client_host, uint32_t client_port);
 
 /**
  * @brief connect to the specified address.
  * @return 0 if OK, negative value if error
  */
 extern int
-nnstreamer_query_connect (uint64_t id, const char *ip, uint32_t port, uint32_t timeout_ms);
+nnstreamer_query_connect (int64_t id, const char *ip, uint32_t port, uint32_t timeout_ms);
 
 /**
  * @brief send command to connected device.
  * @return 0 if OK, negative value if error
  */
 extern int
-nnstreamer_query_send (uint64_t id, TensorQueryCommandData *data, uint32_t timeout_ms);
+nnstreamer_query_send (int64_t id, TensorQueryCommandData *data, uint32_t timeout_ms);
```

Comment:
As we discussed, `tensor_query_client` don't need hash table.
Please replace `int64_t id` as `void * connection`.
And apply it to other APIs commonly used by servers and clients.

Reply:
Updated!